md-setup/src/main/java/com/wipro/ats/bdre/md/setup/SetupDB.java

```diff
@@ -406,6 +411,39 @@ private void populateProcessType(String dataFile) throws MetadataException, IOEx
         }
     }
 
+
+   private void populatePermissionType(String dataFile) throws MetadataException, IOException {
+        String line = null;
+        int lineNum = 0;
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(dataFile));
+            while ((line = br.readLine()) != null) {
+                lineNum++;
+                LOGGER.debug(lineNumber + lineNum + ": " + line);
+                String[] cols = getColumns(line);
+                if (cols == null)
+                    continue;
+                PermissionType pType=new PermissionType();
+                pType.setPermissionTypeId(new Integer(cols[0]));
+                pType.setPermissionTypeName(cols[1]);
+                Object existing = session.get(pType.getClass(), pType.getPermissionTypeId());
+                if (existing == null) {
+                    session.save(pType);
+                }
+            }
+        }catch (MetadataException e) {
+            LOGGER.error(inFile + dataFile + badLine + line);
+            LOGGER.error(e.getMessage());
+            throw new MetadataException(e);
+        }
+        catch (IOException e) {
+            LOGGER.error(inFile + dataFile + badLine + line);
+            LOGGER.error(e.getMessage());
+            throw new IOException(e);
+        }
+    }
+
     private void populateProcessTemplate(String dataFile) throws MetadataException, IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) The Cyclomatic Complexity of this method "populateProcessTemplate" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)